### ISSUE TITLE
switch to pkg-config to find out the information about libpq

### ISSUE THREAD

**Issue by darix** [psycopg/psycopg2#1001]
pg_config was meant for building server extensions.

**Comment by dvarrazzo** [comment]
Patch?

**Comment by dvarrazzo** [comment]
AFAICS pkg-config only deals with the system package. On my ubuntu laptop I have PG 11 installed but libpq.pg declares to be PG 12 (it belongs to the libpq-dev package). I don't think this can be used.

**Comment by darix** [comment]
this is not true. pkg-config support is there since pg 8 or so. in doubt you need to set [PKG_CONFIG_PATH ](https://askubuntu.com/questions/210210/pkg-config-path-environment-variable) to find the .pc files for your local installation. pg_config is not meant for client bindings. 

So I would kindly ask to reopen this bug.

**Comment by darix** [comment]
https://pypi.org/project/pkgconfig/

**Comment by dvarrazzo** [comment]
I can keep it open as a feature request but I'm not keen to dedicate personal time to this change.

**Comment by donaldguy** [comment]
pkg-config would also save some pain for building on macOS with homebrew openssl

I have `export PKG_CONFIG_PATH="/usr/local/opt/openssl/lib/pkgconfig"` in my dotfiles

but that was no dice here.

I had to set an explicit

```export LDFLAGS="-L/usr/local/opt/openssl/lib"
export CPPFLAGS="-I/usr/local/opt/openssl/include"
```

to get psycopg2 to build when everything else has been able to find it

**Comment by darix** [comment]
Does ```pkg-config --libs --cflags openssl``` give you the correct data?

**Comment by dvarrazzo** [comment]
#1315 shows how less maintainable a pkg-config solution is compared to pg_config.
